application/views/survey/questions/answer/gender/buttons/answer.twig

```diff
@@ -13,54 +13,29 @@
 
 <!--Gender question, buttons display -->
 <!-- answer -->
-<ul class="{{ coreClass }} list-unstyled btn-group btn-group-justified" data-toggle="buttons" role="radiogroup" aria-labelledby="ls-question-text-{{basename }}">
+<div class="ls-answers answers-list gender-button btn-group btn-group-justified" role="radiogroup" aria-labelledby="ls-question-text-{{ basename }}">
     <!-- Female -->
-    <li id="javatbd{{ name }}F" class="button-item btn btn-primary {% if fChecked!='' %} active {% endif %}">
-        <input
-            class="radio"
-            type="radio"
-            name="{{ name }}"
-            id="answer{{ name }}F"
-            value="F"
-            {{ fChecked }}
-        />
-        <label for="answer{{ name }}F">
-            <span class="fa fa-venus ls-icon" aria-hidden="true"></span> {{ gT('Female') }}
-        </label>
-    </li>
+
+    <input type="radio" class="btn-check ls-button-radio" name="{{ name }}" id="answer{{ name }}F" value="F" {{ fChecked }}>
+        <label class="btn btn-primary mb-3 answer-item ls-button-label" id="javatbd{{ name }}F" for="answer{{ name }}F">
+        <span class="fa fa-venus ls-icon" aria-hidden="true"></span> {{ gT('Female') }}
+    </label>
 
     <!-- Male -->
-    <li id="javatbd{{ name }}M" class="button-item btn btn-primary  {% if mChecked!='' %} active {% endif %} ">
-        <input
-            class="radio"
-            type="radio"
-            name="{{ name }}"
-            id="answer{{ name }}M"
-            value="M"
-            {{ mChecked }}
-        />
-        <label for="answer{{ name }}M">
-            <span class="fa fa-mars ls-icon" aria-hidden="true"></span> {{ gT('Male') }}
-        </label>
-    </li>
+    <input type="radio" class="btn-check ls-button-radio" name="{{ name }}" id="answer{{ name }}M" value="M" {{ fChecked }}>
+        <label class="btn btn-primary mb-3 answer-item ls-button-label" id="javatbd{{ name }}M" for="answer{{ name }}M">
+        <span class="fa fa-mars ls-icon" aria-hidden="true"></span> {{ gT('Male') }}
+    </label>
 
 <!-- No answer -->
     {% if noAnswer %}
-    <li id="javatbd{{ name }}" class="button-item btn btn-primary  {% if naChecked!='' %} active {% endif %}">
-        <input
-            class="radio"
-            type="radio"
-            name="{{ name }}"
-            id="answer{{ name }}"
-            value=""
-            {{ naChecked }}
-        />
-        <label for="answer{{ name }}">
-            <span class="fa fa-genderless ls-icon" aria-hidden="true"></span> {{ gT('No answer') }}
+        <input type="radio" class="btn-check ls-button-radio" name="{{ name }}" id="answer{{ name }}" value="" {{ naChecked }}>
+        <label class="btn btn-primary mb-3 answer-item ls-button-label" id="javatbd{{ name }}" for="answer{{ name }}">
```

Comment:
Are you sure moving javatbd system to label and NOT tow wrapper is a good idea ?

javatbd is used for hidding by Expression Manager, it's the reaoson of the wrapper …

Reply:
You hidding only label, not input .

gender didn't have array filter, the no issue, but with radio list : you totally broke accessibility …

I really think #javatbd must wrap input + label for accessibility and clearer javascript process.

Reply:
Looks like we're keeping the old structure for radio list, so should work same as before? So if it's only this question type it should be OK?

This is the required HTML structure for Bootstrap 5, so we should stick to it when it's possible, at least.

Reply:
> Looks like we're keeping the old structure for radio list, so should work same as before? So if it's only this question type it should be OK? Especially since there's no way to hide answer options anyway.

> This is the required HTML structure for Bootstrap 5, so we should stick to it when it's possible, at least.

I think you can do with different HTML; like BS3 in fact.

If i don't make error, i update Yes/No button before we release LS 2.50 … not really complex but take time …

Reply:
Compare https://getbootstrap.com/docs/3.3/components/#btn-groups and https://demo.sondages.me/84134?lang=fr#

CSS class do the job.

Reply:
> > Yes, but have same HTML structure is better if we can do it
> 
> I mean, the default must be to do Bootstrap 5 standard, and not hack around to make it look like a previous version. Unless accessibility is affected, then yes.

OK, understandable .